x-pack/plugins/osquery/public/shared_components/osquery_action/use_osquery_menu_item.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiContextMenuItem } from '@elastic/eui';
+import React, { useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { useKibana } from '../../common/lib/kibana';
+
+export const ACTION_OSQUERY = i18n.translate('xpack.osquery.alerts.osqueryAlertTitle', {
+  defaultMessage: 'Run Osquery',
+});
+
+interface IProps {
+  agentId: string;
+  onClick: () => void;
+}
+
+export const useOsqueryMenuItem = ({ agentId, onClick }: IProps) => {
+  const { http } = useKibana().services;
+  const [isAvailable, setIsAvailable] = useState(false);
+  useEffect(() => {
+    (async () => {
+      const result: { item: boolean } = await http.get(
+        `/internal/osquery/fleet_wrapper/agents/${agentId}/available`
```

Comment:
Can you can wrap this in a `try catch`?  What happens if the query fails, should a toaster be surfaced?

Reply:
Sure, wrapper this in try/catch 👍 . I don't see any point in showing a toaster. We just don't show the osquery element in the dropdown.

Reply:
I think there's still a benefit to adding it here in case someone pulls this in somewhere else without wrapping that call in a try catch. Either way I'm fine with it, but just thinking about future proofing a bit...